CONTRIBUTING.md

```diff
@@ -5,70 +5,119 @@ If you have any questions or suggestions we are happy to hear them.
 
-Step 2. Quarkus language server checks its cache if completion options
+**Step 2.** MicroProfile language server checks its cache if completion options
 exist.  
-* If it exists the Quarkus language server sends them to VSCode 
+* If it exists the MicroProfile language server sends them to VSCode 
 as the response to the `textDocument/completion` request.
 Communication is complete, and does not proceed to Step 3 
 and onwards.
-* If it does not exist, the Quarkus language server sends a 
-custom request, `quarkus/projectInfo` to vscode-quarkus.  
+* If it does not exist, the MicroProfile language server sends a 
+custom request, `microprofile/projectInfo` to vscode-quarkus.  
 Proceed to Step 3.  
 
-Step 3. vscode-quarkus receives the `quarkus/projectInfo` request,
-and runs the `quarkus.java.projectInfo` command. This command was
-defined by the Quarkus jdt.ls extension.  
+**Step 3.** vscode-quarkus receives the `microprofile/projectInfo` request,
+and delegates it to the MicroProfile jdt.ls extension.  
 
-Step 4. The Quarkus jdt.ls extension receives the command, determines
-information about the currently opened Quarkus project and returns
-the information to vscode-quarkus.  
+**Step 4.** The MicroProfile jdt.ls extension receives the command, determines
+[project information](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/commons/MicroProfileProjectInfo.java)
+(project URI, configuration properties, hints etc.)
+about the currently opened (MicroProfile or Quarkus) project and returns
+the information to vscode-quarkus. The project information is then sent to
+the MicroProfile language server
 
-The information is computed by scanning all classes annotated with `@ConfigRoot` Quarkus annotation from all project classpath JARs and generating the proper Quarkus properties.
+**Step 5.** vscode-quarkus receives the project information and sends it
+to the MicroProfile language server.  
 
-For instance if the project has a `pom.xml` with this dependency:
+**Step 6.** MicroProfile language server receives the information, adds it 
+to its cache, and returns the completion options stored in the 
+project information as the response to the `textDocument/completion`
+request.
+
+## Implementing language features for `application.properties`
+When an `application.properties` file sends a request (e.g. textDocument/completion) to the
+MicroProfile language server, the requests are accepted in
+[ApplicationPropertiesTextDocumentService#completion](https://github.com/redhat-developer/quarkus-ls/blob/9eb7718a6e0faa300f3937e607c3bfffc25c2f1d/microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/ls/ApplicationPropertiesTextDocumentService.java#L150).
+This class receives LSP requests for `applicatn't exist, when project dependencies chan't exist, when project dependencies chaon.properties` files.
```

Comment:
please fix typo with `applicatn't `and `chan't`

Reply:
Thanks for catching this. Fixed